### PR TITLE
add u.omitted constant

### PR DESCRIPTION
**Pull request by yanick**
Right now if one wants to remove properties and update others, we need to do it in two steps: one call to `u.omit` and one to `u.update`. This PR adds a `u.omitted` constant signalling that the property should be omitted (we can't use `null` or `undefined` as those could be legit property values we might want to keep).

### CODE REVIEW

File: lib/update.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import isPlainObject from 'lodash/isPlainObject'
+import _omitBy from 'lodash/omitBy'
+
 import wrap from './wrap'
+import constant from './constant'
+
+const innerOmitted = constant('omitted')
```
Comment (aaronjensen) [inline lib/update.js]: I think this can just be `{}` since that will work for reference equality. Making it a function that returns the string `omitted` doesn't seem necessary unless I'm missing something.

File: lib/update.js [hunk 2]
```diff
@@ -19,7 +25,9 @@ function resolveUpdates(updates, object) {
       let updatedValue = value
 
       if (
-        !Array.isArray(value) && value !== null && typeof value === 'object'
+        !Array.isArray(value) &&
```
Comment (aaronjensen) [inline lib/update.js]: I'm down with prettiering the repo, but I'd prefer it be a separate commit/PR. Could you please amend any of the non-essential changes out?

File: test/updeep-spec.js [hunk 3]
```diff
@@ -142,4 +142,13 @@ describe('updeep', () => {
...
   })
+
+    it('omit properties via u.omitted', () => {
+        const result = u({
+            a: u.omitted,
+            b: i => i + 1,
+        }, { a: 1, b: 2 })
+
+        expect(result).to.eql({ b: 3 });
+    })
```
Comment (aaronjensen) [inline test/updeep-spec.js]: Please include a test for a nested object and for an array (what should it do if you omit in an array?)

File: README.md [hunk 4]
```diff
@@ -394,6 +394,18 @@ var user = {
...
+
+### `u.omitted`
+
+A property updated to this constant will be removed from the final object.
+Useful when one wishes to remove and update properties in a single operation.
+
+```js
+var user = { email: 'john@aol.com', username: 'john123', authToken: '1211..' };
+
+var result = u({ authToken: _.omitted, active: true }, user);
```
[the remark this answers is not in the document]
Comment (yanick) [reply README.md]: urgh. u.cantbelievehowoftenImakethatypo

File: lib/update.js (hunk 1 above)
Comment (aaronjensen) [reply lib/update.js]: Perhaps something like `{ __omitted: true }`. Still a reference, just more meaningful if seen in debug

Comment (aaronjensen) [reply lib/update.js]: Also, there are some dangers to relying on reference. If a person has two versions of updeep in their tree, it's possible that this won't work properly. Maybe it should be modeled after however lodash does their placeholder? I don't remember how off the top of my head.

File: test/updeep-spec.js (hunk 3 above)
Comment (yanick) [reply test/updeep-spec.js]: tests added.

The `_.omitted` should skip the property value, no matter if it's a scalar or an array or object, so we should be good there.

The undefined case is if one does:

```
u({ 3: u.omitted })([ 1,2,3,4,5 ]);
```

I could add logic to take care of that case, basically do

```
  if (Array.isArray(defaultedObject)) {
    return updateArray(resolvedUpdates, defaultedObject).filter( item => item !== innerOmitted )
  }
```

 but it just doesn't feel like something anybody would do.

File: lib/update.js (hunk 1 above)
Comment (yanick) [reply lib/update.js]: [lodash] Hmmm... I'll have a look.

File: test/updeep-spec.js (hunk 3 above)
Comment (aaronjensen) [reply test/updeep-spec.js]: Yeah, I guess the question is, what should the lib do... throw? Create a sparse array? filter it out as you demonstrate there? 

Maybe throwing for now in `updateArray` would work.

File: lib/update.js (hunk 1 above)
Comment (yanick) [reply lib/update.js]: Although I think the ref won't work only if somebody does:

```
import u1 from 'updeep';
import u2 from './path/to/other/updeep';

u1({ foo: u2.omitted })(x);
```

which would be rather twisted.

Comment (aaronjensen) [reply lib/update.js]: agree that it's unlikely, if someone has a library that depends on updeep to build custom operators you could get into this situation. That library may export an operator that uses `omitted` from its version of `updeep` and if you use it with yours...

Not likely, I agree, but if lodash has an easy way to deal w/ it, it's probably good to follow that pattern.

Comment (yanick) [reply lib/update.js]: And unless I'm mistaken, lodash does it by attaching a `placeholder` attribute to the main exported `_` variable: https://github.com/lodash/lodash/blob/4ea8c2ec249be046a0f4ae32539d652194caf74f/.internal/getHolder.js

We should be able to do the same. Give me 5 minutes. :-)

Comment (yanick) [reply lib/update.js]: Darn. `update` is wrapped via `wrap` before it's exported, so binding would be a little more involved than I thought. And I'm not sure it'd bring any more safety than using the `omitted` and `update` from the same `u` source.

Comment (aaronjensen) [reply lib/update.js]: okay, let's just try a reference for now.

**Comment by aaronjensen** [comment]
looks like there are some formatting errors, could you run prettier/fix those up, please?

**Comment by yanick** [comment]
done!

**Comment by aaronjensen** [comment]
Could you rebase on master when you get a chance please? I think that'll fix the build.

**Comment by yanick** [comment]
done!

**Comment by aaronjensen** [comment]
Looks great, thank you for your contribution!

**Comment by aaronjensen** [comment]
Released as 1.1.0, thanks again.

**Comment by yanick** [comment]
Pleasure was all mine! Thanks! :-)
